### PR TITLE
[GLUTEN-1472][CH] Refine clickhouse memory audit

### PR DESCRIPTION
**Pull request by binmahone**
this PR is first step of https://github.com/oap-project/gluten/issues/1472

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -294,6 +294,12 @@
           </exclusion>
         </exclusions>
       </dependency>
+      <dependency>
+        <groupId>org.apache.spark</groupId>
+        <artifactId>spark-yarn_${scala.binary.version}</artifactId>
```
Comment (zzcclp) [inline pom.xml:299]: why it needs to add spark-yarn jar?

Comment (binmahone) [reply pom.xml:299]: it's for testing on yarn. it's provided so it won't incur overhead

**Comment by binmahone** [comment]
retest this please

**Comment by binmahone** [comment]
test this please

**Comment by binmahone** [comment]
Run Gluten Clickhouse CI

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**